eclass: attempt to remove icedtea{,-bin} from eclasses

vaukai opened the pull request:
@fordfrog 
going though the eclasses for cleanup from the last rited `dev-java/icedtea{,-bin}` i am not sure, could we even drop the whole [java-pkg_setup-vm()](https://github.com/gentoo/gentoo/blob/master/eclass/java-utils-2.eclass#L2632-L2644) and [java-pkg_get-vm-vendor()](https://github.com/gentoo/gentoo/blob/master/eclass/java-utils-2.eclass#L2672-L2681) or even more?

also, i am getting confused about `java-pkg_get-current-vm()` which seems to appear twice in java-utils-2.eclass:

```
java-pkg_get-current-vm() {
	echo ${GENTOO_VM}
}
```

```
# @FUNCTION: java-pkg_get-current-vm
# @INTERNAL
# @RETURN: The current VM being used
java-pkg_get-current-vm() {
	java-config -f
}
```

fordfrog commented:
> @fordfrog going though the eclasses for cleanup from the last rited `dev-java/icedtea{,-bin}` i am not sure, could we even drop the whole [java-pkg_setup-vm()](https://github.com/gentoo/gentoo/blob/master/eclass/java-utils-2.eclass#L2632-L2644) and [java-pkg_get-vm-vendor()](https://github.com/gentoo/gentoo/blob/master/eclass/java-utils-2.eclass#L2672-L2681) or even more?

technically those are not needed anymore, but we still have https://github.com/gentoo/gentoo/pull/35942 pending though there was no update after icedtea removal so there is a chance `icedtea` will return back to the tree, but for that we need to have some feedback from @gnu-andrew.

> also, i am getting confused about `java-pkg_get-current-vm()` which seems to appear twice in java-utils-2.eclass:
> 
> ```
> java-pkg_get-current-vm() {
> 	echo ${GENTOO_VM}
> }
> ```
> 
> ```
> # @FUNCTION: java-pkg_get-current-vm
> # @INTERNAL
> # @RETURN: The current VM being used
> java-pkg_get-current-vm() {
> 	java-config -f
> }
> ```

that's interesting. i'm not sure which one is used, that's what's the precedence in the eclasses, whether the first function is overriden by the second one or the second one is ignored. but one function should be enough. i just also checked the code of `java-config` and it seems to use `GENTOO_VM` too to get the active vm. and as `GENTOO_VM` is used quite a lot in the eclass, we can use that function instead of calling `java-config`.

vaukai commented:
> that's interesting. i'm not sure which one is used, that's what's the precedence in the eclasses, whether the first function is overriden by the second one or the second one is ignored. but one function should be enough. i just also checked the code of `java-config` and it seems to use `GENTOO_VM` too to get the active vm. and as `GENTOO_VM` is used quite a lot in the eclass, we can use that function instead of calling `java-config`.

some tests i did showed that the first one was never used unless i removed the second one. so we could remove the second if there wasn't [this section](https://github.com/gentoo/gentoo/blob/master/eclass/java-utils-2.eclass#L2769-L2775) telling it uses `java-config -f`.  and also, `java-config` is used very often in this eclass too. 
so what about removing the presently unused first one?

fordfrog commented:
> some tests i did showed that the first one was never used unless i removed the second one. so we could remove the second if there wasn't [this section](https://github.com/gentoo/gentoo/blob/master/eclass/java-utils-2.eclass#L2769-L2775) telling it uses `java-config -f`. and also, `java-config` is used very often in this eclass too. so what about removing the presently unused first one?

pick anyone you want. functionally they are the same, and as both variants are used in the eclass, there's no preference. it would be nice to use only one approach across the whole eclass, but idk whether it's worth the effort now.

vaukai commented:
> > some tests i did showed that the first one was never used unless i removed the second one. so we could remove the second if there wasn't [this section](https://github.com/gentoo/gentoo/blob/master/eclass/java-utils-2.eclass#L2769-L2775) telling it uses `java-config -f`. and also, `java-config` is used very often in this eclass too. so what about removing the presently unused first one?
> 
> pick anyone you want. functionally they are the same, and as both variants are used in the eclass, there's no preference. it would be nice to use only one approach across the whole eclass, but idk whether it's worth the effort now.

Added to https://github.com/gentoo/gentoo/pull/35287
